ddl/ddl_test.go

```diff
@@ -138,37 +143,45 @@ type historyJobArgs struct {
 	tblIDs map[int64]struct{}
 }
 
-func checkEqualTable(c *C, t1, t2 *model.TableInfo) {
-	c.Assert(t1.ID, Equals, t2.ID)
-	c.Assert(t1.Name, Equals, t2.Name)
-	c.Assert(t1.Charset, Equals, t2.Charset)
-	c.Assert(t1.Collate, Equals, t2.Collate)
-	c.Assert(t1.PKIsHandle, DeepEquals, t2.PKIsHandle)
-	c.Assert(t1.Comment, DeepEquals, t2.Comment)
-	c.Assert(t1.AutoIncID, DeepEquals, t2.AutoIncID)
+func checkEqualTable(t *testing.T, t1, t2 *model.TableInfo) {
+	require.Equal(t, t1.ID, t2.ID)
+	require.Equal(t, t1.Name, t2.Name)
+	require.Equal(t, t1.Charset, t2.Charset)
+	require.Equal(t, t1.Collate, t2.Collate)
+	if !reflect.DeepEqual(t1.PKIsHandle, t2.PKIsHandle) {
+		t.FailNow()
+	}
+	if !reflect.DeepEqual(t1.Comment, t2.Comment) {
+		t.FailNow()
+	}
+	if !reflect.DeepEqual(t1.AutoIncID, t2.AutoIncID) {
+		t.FailNow()
+	}
 }
 
-func checkHistoryJob(c *C, job *model.Job) {
-	c.Assert(job.State, Equals, model.JobStateSynced)
+func checkHistoryJob(t *testing.T, job *model.Job) {
+	require.Equal(t, job.State, model.JobStateSynced)
 }
 
-func checkHistoryJobArgs(c *C, ctx sessionctx.Context, id int64, args *historyJobArgs) {
+func checkHistoryJobArgs(t *testing.T, ctx sessionctx.Context, id int64, args *historyJobArgs) {
 	txn, err := ctx.Txn(true)
-	c.Assert(err, IsNil)
-	t := meta.NewMeta(txn)
-	historyJob, err := t.GetHistoryDDLJob(id)
-	c.Assert(err, IsNil)
-	c.Assert(historyJob.BinlogInfo.FinishedTS, Greater, uint64(0))
+	require.NoError(t, err)
+	trans := meta.NewMeta(txn)
+	historyJob, err := trans.GetHistoryDDLJob(id)
+	require.NoError(t, err)
+	require.Greater(t, historyJob.BinlogInfo.FinishedTS, uint64(0))
 
 	if args.tbl != nil {
-		c.Assert(historyJob.BinlogInfo.SchemaVersion, Equals, args.ver)
-		checkEqualTable(c, historyJob.BinlogInfo.TableInfo, args.tbl)
+		require.Equal(t, historyJob.BinlogInfo.SchemaVersion, args.ver)
+		checkEqualTable(t, historyJob.BinlogInfo.TableInfo, args.tbl)
 		return
 	}
 
 	// for handling schema job
-	c.Assert(historyJob.BinlogInfo.SchemaVersion, Equals, args.ver)
-	c.Assert(historyJob.BinlogInfo.DBInfo, DeepEquals, args.db)
+	require.Equal(t, historyJob.BinlogInfo.SchemaVersion, args.ver)
+	if !reflect.DeepEqual(historyJob.BinlogInfo.DBInfo, args.db) {
```

Comment:
`require.Equal` is bitwise equality checks. BTW this file is also not under `ddl/table`. We break down subtasks for small changes easy to review and merge, and then repeat it for the next iteration.

From https://github.com/pingcap/tidb/pull/29288#issuecomment-955878811

> It's totally OK to touch multiple files. Never mind. But yes please keep necessary changes only :)

Please notice **please keep necessary changes only**.

Reply:
Thanks! Changed it to require.Equal.